example/lib/develop/peripheral_list/peripheral_list_bloc.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:async';
+import 'package:blemulator_example/develop/model/ble_peripheral.dart';
+import 'package:blemulator_example/develop/repository/peripheral_list_repository.dart';
+import 'package:bloc/bloc.dart';
+import './bloc.dart';
+
+class PeripheralListBloc
+    extends Bloc<PeripheralListEvent, PeripheralListState> {
+  List<BlePeripheral> _peripherals = <BlePeripheral>[];
+  bool _scanningEnabled = false;
+  PeripheralListRepository _peripheralListRepository;
+
+  PeripheralListBloc(this._peripheralListRepository);
+
+  @override
+  PeripheralListState get initialState =>
+      PeripheralListState(_peripherals, _scanningEnabled);
+
+  @override
+  Stream<PeripheralListState> mapEventToState(
+    PeripheralListEvent event,
+  ) async* {
+    if (event is StartPeripheralScan) {
+      yield* _mapStartPeripheralScanToState(event);
+    } else if (event is StopPeripheralScan) {
+      yield* _mapStopPeripheralScanToState(event);
+    } else if (event is NewPeripheralScan) {
+      yield* _mapNewPeripheralScanToState(event);
+    } else if (event is PickPeripheral) {
+      // TODO: Logic to pick a device
+    } else if (event is ConnectToPeripheral) {
+      // TODO: Logic to connect to device
+    } else if (event is DisconnectFromPeripheral) {
+      // TODO: Logic to disconnect from device
+    }
+  }
+
+  Stream<PeripheralListState> _mapStartPeripheralScanToState(
+      StartPeripheralScan event) async* {
+    _peripheralListRepository.startPeripheralScan(
+        scanEventOutput: (BlePeripheral peripheral) {
+      add(NewPeripheralScan(peripheral));
+    });
+    _setScanningEnabled(true);
+    yield PeripheralListState(_peripherals, _scanningEnabled);
+  }
+
+  Stream<PeripheralListState> _mapStopPeripheralScanToState(
+      StopPeripheralScan event) async* {
+    await _peripheralListRepository.stopPeripheralScan();
+    _setScanningEnabled(false);
+    _clearPeripherals();
+    yield PeripheralListState(_peripherals, _scanningEnabled);
+  }
+
+  Stream<PeripheralListState> _mapNewPeripheralScanToState(
+      NewPeripheralScan event) async* {
+    if (!_peripherals.contains(event.peripheral)) {
```

Comment:
on what basis the devices are compared?

Reply:
`BlePeripheral` class extends `Equatable` from `equatable` plugin, which effectively is responsible for overriding `==` operator and `hashCode` property, by the use of `props` method which in `BlePeripheral`s case returns the list `[name, id]`.

So to sum this up - two peripherals are considered to be equal when their `name` and `id` are equal.